torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -788,6 +791,258 @@ def propagate_meta_info(new_hop_gm, new_call_function_node, old_call_function_no
+        # Outputs symbolic scalars, tensors  are accumulated separately.
+        # Then in forward outputs and backward inputs installed in order
+        # sym_scalars, packed_saved_tensors.
+        # Keeping all tensors together allows to preserve
+        # the same identification at runtime,
+        # updating only number of saved sym_scalars and tensors.
+        pack_g_inputs = pack_g.find_nodes(op="placeholder")
+        assert len(pack_g_inputs) == 1
+        env = {pack_g_inputs[0]: saved}
+        fw_pack_out_n = None
+        with fw_g.inserting_before(fw_out_n):
+            for node in pack_g.nodes:
+                if node.op == "placeholder":
+                    continue
+                new_n = fw_g.node_copy(node, lambda n: env[n])
+                env[node] = new_n
+                # Output node is temporarily copied to have remapped arguments.
+                # Removed in the end.
+                if node.op == "output":
+                    fw_pack_out_n = new_n
+        env.clear()
+        assert fw_pack_out_n
+        for n in pytree.tree_leaves(fw_pack_out_n.args[0]):
+            if not isinstance(n, torch.fx.Node):
+                continue
+
+            if isinstance(n.meta["val"], torch.Tensor):
+                fw_outs_packed_tensors.append(n)
+            elif is_sym_node(n):
+                fw_outs_packed_syms.append(n)
+        fw_g.erase_node(fw_pack_out_n)
+
+        # Install unpack hook graph as a prologue of backward graph
+        # Saved tensors inputs are replaced with packed tensors and packed sym scalars.
+        # The saved tensors inputs usages in the graph are replaced with unpack hook graph outputs.
+        unpack_gm = prepare_hook_gm(aot_config, unpack_hook_gm, (pack_out_val,))
+        unpack_g = unpack_gm.graph
+        maybe_log_graph(
+            unpack_gm,
+            f"saved_tensors_unpack_hook {saved.name}",
+            "aot_saved_tensors_hooks_unpack",
+            aot_config,
+        )
+
+        def find_saved_in_bw_inputs(bw_inputs):
+            for n in bw_inputs:
+                if n.name == saved.name:
+                    return n
+
+        bw_g_input = find_saved_in_bw_inputs(bw_g_inputs)
+        assert bw_g_input
+        # Replace backward graph saved tensor input with copy of pack graph outputs
+        # All non-Tensor, non-symscalars outputs are constanted.
+
+        unpack_g_inputs = unpack_g.find_nodes(op="placeholder")
+        env = {}
+        num_saved_for_bw = len(fw_outs_saved_for_bw)
+        for unp_in_n, val in zip(unpack_g_inputs, pytree.tree_leaves(pack_out_val)):
+            is_sym = isinstance(val, py_sym_types)
+            if isinstance(val, torch.Tensor) or is_sym:
+                new_node_name = bw_g_input.name + "_" + unp_in_n.name
+                # Backward calling convention: ctx_symints,ctx_saved_tensors
+                # Inserting packed sym scalars before first saved tensor input.
```

Comment:
hmm can you convince me that the sym scalar handling is robust? Or otherwise we can leave it as unsupported for now too

Reply:
There is a test that works fine.
I have not found any problems with it.
I can add more tests :)